syntax = "proto3";

message peak_p
{
	double mz_value = 1;
	double intensity = 2;
}

message spectra_p
{
	string file_name = 1;
	string spectra_name = 2;

	repeated peak_p ions = 3;
}

message ms_data 
{
	repeated int32 idxes = 1;

	repeated spectra_p data = 2;
}

message assignment
{
	int32 num_proc = 1;		// to identify the num_proc
	
	repeated ms_data multi_dataset = 2;		// distributed datasets
}
